CONTRIBUTING.md

```diff
@@ -1,18 +1,126 @@
+4. Wait for input of others, even if you could implement it right away.  It is always suggested to _first_ suggest a quest and implement it only if it has been accepted there.
+
+➡️ If you can code, see also how to [develop your own quest](#developing-new-quests).
+
+A quest is officially marked as accepted, when it get's the ["new quest" tag](https://github.com/westnordost/StreetComplete/labels/new%20quest).
+<!-- TODO: introduce a better way/new tag here? ref https://github.com/westnordost/StreetComplete/issues/1006 -->
+
+## Improving documentation
+
+- ☑️ requires no programming knowledge
+- ⬜️ requires no OpenStreetMap knowledge
+
+You can also help to keep the OpenStreetMap wiki **up-to-date** for StreetComplete. For example:
+* Add missing quests in [the quest list](https://wiki.openstreetmap.org/wiki/StreetComplete/Quests) or check, that they are up to date.
+* Edit [the FAQ](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ) and add reoccuring questions.
+* Edit [the main StreetComplete page](https://wiki.openstreetmap.org/wiki/StreetComplete).
+
+## Development
+
+- ⬜️ requires no programming knowledge
+- ☑️ requires no OpenStreetMap knowledge (depends on what you do)
+
+If you would like to help and are able to contribute code, you are most welcome.
+
+There are many reasonable feature requests and ideas for new question types in the issue tracker which you could also engage yourself with. If you have own ideas how to improve this app and want to make sure that the Pull Request will be merged, it is strongly suggested to **open an issue first** to discuss the feature.
+
+Note StreetComplete also uses [some dependencies](#issues-of-dependencies), where contributions are likely also accepted and help StreetComplete, too.
+
+If you need to find things, where help is appreciated [have a look at the issues tagged with "help wanted"](https://github.com/westnordost/StreetComplete/labels/help%20wanted).
+
+### Developing new quests
+
+You want to contribute a new quest right away? That's great!
+However, we strongly suggest to [**open an issue** discussing the quest](#suggesting-new-quests), before creating a PR, so we can see if your quest idea meets the criteria and can be included in StreetComplete.
+
+In case you **don't have an idea of a quest?** Look [at the existing issues](https://github.com/westnordost/StreetComplete/issues?q=is%3Aissue+is%3Aopen+label%3A%22new+quest%22+sort%3Areactions-%2B1-desc). Sorted by 👍 reactions you can also see which are the most requested quests.
+
+To start developing you can just download [Android Studio](https://developer.android.com/studio/index.html) and create your quest. It should automatically handle indentation through the [editorconfig file](.editorconfig). If you use another IDE, you may need to [install a plugin](http://editorconfig.org/#download).
```

Comment:
I know nobody who uses anything else than Android Studio for Android development. And those who do, will know their way around.

Reply:
Yeah, I agree this part is quite rare. What would also suit here may be general coding guidelines. However, I can hardly invent them, so this is something you have to decide/write down. Just explain the style/set rules on how things should be written. (one I may imagine is e.fg. "abstract everything, which is possible" or so).

I'll add a TODO in the code for that.